melodies_monet/plots/aircraftplots.py

```diff
@@ -0,0 +1,842 @@
+        (e.g., color, linestyle, markerstyle).
+    fig_dict : dictionary
+        Dictionary containing information about the figure.
+    text_dict : dictionary
+        Dictionary containing information about text.
+    debug : bool
+        Whether to plot interactively (True) or not (False). Flag for submitting jobs to supercomputer turn off interactive mode.
+    interquartile_style= str
+        Whether the vertical profile uses shading or box style for interquartile range
+        
+    Returns
+    -------
+    ax : ax
+        Matplotlib ax such that driver.py can iterate to overlay multiple models on the same plot.
+    """
+    if debug == False:
+        plt.ioff()
+    
+    # First, define items for all plots
+    # Set default text size
+    def_text = dict(fontsize=14)
+    if text_dict is not None:
+        text_kwargs = {**def_text, **text_dict}
+    else:
+        text_kwargs = def_text
+    
+    # Set ylabel to column if not specified
+    if ylabel is None:
+        ylabel = column
+    if label is not None:
+        plot_dict['label'] = label
+    if vmin is not None and vmax is not None:
+        plot_dict['ylim'] = [vmin, vmax]
+    # Scale the fontsize for the x and y labels by the text_kwargs
+    plot_dict['fontsize'] = text_kwargs['fontsize'] * 0.8
+
+      
+                         
+    # Then, if no plot has been created yet, create a plot and plot the obs
+    if ax is None: 
+        # First define the colors for the observations
+        obs_dict = dict(color='k', linestyle='-', marker='*', linewidth=1.2, markersize=6.)
+        if plot_dict is not None:
+            # Whatever is not defined in the yaml file is filled in with the obs_dict here
+            plot_kwargs = {**obs_dict, **plot_dict} 
+        else:
+            plot_kwargs = obs_dict
+        # Create the figure
+        if fig_dict is not None:
+            f, ax = plt.subplots(**fig_dict)    
+        else: 
+            f, ax = plt.subplots(figsize=(10, 6))
+       
+        # Bin the altitude variable and calculate median and interquartiles
+        altitude_bins = pd.cut(df[altitude_variable], bins=bins)
+        # Calculate the midpoints of the altitude bins
+        bin_midpoints = altitude_bins.apply(lambda x: x.mid)
+        # Convert bin_midpoints to a column in the DataFrame
+        df['bin_midpoints'] = bin_midpoints
+        median = df.groupby(altitude_bins)[column].median()
+        q1 = df.groupby(altitude_bins)[column].quantile(0.25)
+        q3 = df.groupby(altitude_bins)[column].quantile(0.75)
+        # Convert bin_midpoints to a numerical data type
+        df['bin_midpoints'] = df['bin_midpoints'].astype(float)
+
+        p5 = df.groupby(altitude_bins)[column].quantile(0.05)
+        p10 = df.groupby(altitude_bins)[column].quantile(0.10)
+        p90 = df.groupby(altitude_bins)[column].quantile(0.90)
+        p95 = df.groupby(altitude_bins)[column].quantile(0.95)
```

Comment:
`p5` and `p95` are never used.

```suggestion
        p10 = df.groupby(altitude_bins)[column].quantile(0.10)
        p90 = df.groupby(altitude_bins)[column].quantile(0.90)
```